src/service.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  AutosuggestClient,
+  AvailableLanguagesClient,
+  ConvertTo3waClient,
+  ConvertToCoordinatesClient,
+  GridSectionClient,
+} from './client';
+import type {
+  AutosuggestResponse,
+  AutosuggestSuggestion,
+  AutosuggestOptions,
+  AvailableLanguagesResponse,
+  ConvertTo3waOptions,
+  ConvertToCoordinatesOptions,
+  GridSectionOptions,
+  GridSectionResponse,
+  LocationGeoJsonResponse,
+  LocationJsonResponse,
+} from './client';
+import { ApiClientConfiguration, fetchTransport, axiosTransport } from './lib';
+import type { Transport } from './lib';
+
+export type What3wordsService = {
+  clients: {
+    autosuggest: AutosuggestClient;
+    availableLanguages: AvailableLanguagesClient;
+    convertTo3wa: ConvertTo3waClient;
+    convertToCoordinates: ConvertToCoordinatesClient;
+    gridSection: GridSectionClient;
+  };
+  setApiKey: (key: string) => void;
+  setConfig: (config: ApiClientConfiguration) => void;
+  autosuggest: (options: AutosuggestOptions) => Promise<AutosuggestResponse>;
+  autosuggestSelection: (options: AutosuggestSuggestion) => Promise<void>;
+  availableLanguages: () => Promise<AvailableLanguagesResponse>;
+  convertTo3wa: (
+    options: ConvertTo3waOptions
+  ) => Promise<LocationGeoJsonResponse | LocationJsonResponse>;
+  convertToCoordinates: (
+    options: ConvertToCoordinatesOptions
+  ) => Promise<LocationJsonResponse | LocationGeoJsonResponse>;
+  gridSection: (options: GridSectionOptions) => Promise<GridSectionResponse>;
+};
+
+export function what3words(
+  apiKey?: string,
+  config?: ApiClientConfiguration,
+  opts?: { transport: 'fetch' | 'axios' | Transport }
+): What3wordsService {
+  const transport =
+    typeof opts?.transport === 'function'
+      ? opts?.transport
+      : opts?.transport === 'axios'
+      ? axiosTransport()
+      : fetchTransport();
```

Comment:
```suggestion
 function getTransport(opts: { transport: 'fetch' | 'axios' | Transport }){
   const isAxiosTransport = typeof opts?.transport === 'function'
   ? opts?.transport
   : opts?.transport === 'axios'
   return isAxiosTransport ? axiosTransport() : fetchTransport()
 }
  const transport = getTransport()
```

Reply:
This actually wouldn't work, because the case where `typeof opts.transport === 'function'` would not return the custom transport function. I have moved this to a function though as suggestion with the correct implementation.